src/charm.py

```diff
@@ -157,6 +158,21 @@ def _on_get_issued_certificates(self, event: ActionEvent) -> None:
         results = {"certificates": [certificate.to_json() for certificate in certificates]}
         event.set_results(results)
 
+    def _on_rotate_private_key(self, event: ActionEvent):
+        """Handle the rotate-private-key action.
+
```

Comment:
We can probably improve the docstring, something similar to the content in charmcraft.yaml would be useful

Reply:
Good point, done